src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java

```diff
@@ -0,0 +1,26 @@
+package com.codeless.api.automation.storage;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class RateLimitStorage {
```

Comment:
if you want you inject redis client here and store that data in Redis. As of now, it is ok, but It is good to have redis here.

Reply:
Yep! As wee agreed - Redis will be follow up step.